experimental/generator-dotnet-yeoman/generators/app/index.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const Generator = require('yeoman-generator');
+const path = require('path');
+const { v4: uuidv4 } = require('uuid');
+
+const INTEGRATION_WEBAPP = 'webapp';
+const INTEGRATION_FUNCTIONS = 'functions';
+
+const PLATFORM_DOTNET = 'dotnet';
+const PLATFORM_JS = 'js';
+const PLATFORM_JAVA = 'java';
+const PLATFORM_PYTHON = 'python';
+
+const PROJECT_TYPEID_WEBAPP = '9A19103F-16F7-4668-BE54-9A1E7A4F7556';
+const PROJECT_TYPEID_FUNCTION = 'FAE04EC0-301F-11D3-BF4B-00C04F79EFBC';
+
+module.exports = class extends Generator {
+    constructor(args, opts) {
+        super(args, opts);
+
+        this.argument('botName', {
+            type: String,
+            required: true
+        });
+
+        this.option('integration', {
+            desc: `The host integration to use:  ${INTEGRATION_WEBAPP} or ${INTEGRATION_FUNCTIONS}`,
+            type: String,
+            default: INTEGRATION_WEBAPP,
+            alias: 'i'
+        });
+
+        this.option('platform', {
+            desc: `The programming platform to use: ${PLATFORM_DOTNET}`,
+            type: String,
+            default: PLATFORM_DOTNET,
+            alias: 'p'
+        });
+
+        this._verifyOptions();
+    }
+
+    _verifyOptions() {
+        if (this.options.integration != INTEGRATION_WEBAPP && this.options.integration != INTEGRATION_FUNCTIONS) {
+            throw new Error(`--integration must be: ${INTEGRATION_WEBAPP} or ${INTEGRATION_FUNCTIONS}`);
+        }
+
+        if (this.options.platform != PLATFORM_DOTNET) {
```

Comment:
Use === or !== for equality comparisons (the extra = ensures the type is the same as well as the value, otherwise something like 2 == '2' would equate to true).